app_dart/test/request_handlers/github_webhook_test.dart

```diff
@@ -272,6 +272,116 @@ void main() {
+            contains('a: accessibility'));
+        expect(GithubWebhook.getLabelsForFrameworkPath('foo/bar/baz/accessibility/voiceover.dart'),
+            contains('a: accessibility'));
+      });
+
+      test('Examples label applied', () {
+        expect(GithubWebhook.getLabelsForFrameworkPath('examples/foo/bar/baz.dart'), contains('d: examples'));
+      });
+
+      test('Gallery label applied', () {
+        expect(GithubWebhook.getLabelsForFrameworkPath('examples/flutter_gallery/lib/gallery.dart'),
+            contains('team: gallery'));
+      });
+
+      test('Team label applied', () {
+        expect(GithubWebhook.getLabelsForFrameworkPath('dev/foo/bar/baz.dart'), contains('team'));
+        expect(GithubWebhook.getLabelsForFrameworkPath('examples/foo/bar/baz.dart'), contains('team'));
+      });
+    });
+
+    group('getLabelsForEnginePath', () {
+      test('No label is applied to paths with no applicable label', () {
+        expect(GithubWebhook.getLabelsForEnginePath('nonsense/path/foo.cc'), isEmpty);
+      });
+
+      test('platform-android applied for Android embedder', () {
+        expect(GithubWebhook.getLabelsForEnginePath('shell/platform/android/RIsForRhubarbPie.java'),
+            contains('platform-android'));
+      });
+
+      test('platform-ios and platform-macos applied for common Darwin code', () {
+        expect(GithubWebhook.getLabelsForEnginePath('shell/platform/darwin/common/ThinkDifferent.mm'),
+            containsAll(<String>['platform-ios', 'platform-macos']));
+      });
+
+      test('platform-ios and platform-ios applied for iOS embedder', () {
+        expect(
+            GithubWebhook.getLabelsForEnginePath('shell/platform/darwin/ios/BackButton.mm'), contains('platform-ios'));
+      });
+
+      test('platform-macos applied for macOS embedder', () {
+        expect(GithubWebhook.getLabelsForEnginePath('shell/platform/darwin/macos/PhysicalEscapeKey.mm'),
+            contains('platform-macos'));
+      });
+
+      test('platform-fuchsia applied for fuchsia embedder', () {
+        expect(GithubWebhook.getLabelsForEnginePath('shell/platform/fuchsia/spell_checker.cc'),
+            contains('platform-fuchsia'));
+      });
+
+      test('platform-linux applied for linux embedder', () {
+        expect(GithubWebhook.getLabelsForEnginePath('shell/platform/linux/systemd_integration.cc'),
+            contains('platform-linux'));
+      });
+
+      test('platform-windows applied for windows embedder', () {
+        expect(
+            GithubWebhook.getLabelsForEnginePath('shell/platform/windows/start_menu.cc'), contains('platform-windows'));
+      });
+
+      test('platform-web applied for web paths', () {
+        expect(GithubWebhook.getLabelsForEnginePath('lib/web_ui/shadow_dom.dart'), contains('platform-web'));
+        expect(GithubWebhook.getLabelsForEnginePath('web_sdk/'), contains('platform-web'));
+      });
+    });
```

Comment:
Up to you, but you could deduplicate some of the boiler plate by doing a table driven test

Reply:
True but table-driven tests aren't too idiomatic in Dart, and are likely to be a little harder to debug, though certainly more concise. If you can think of a way to do it that'll fail with an exactly line-number etc. though, definitely go for it!